### PR TITLE
WIP of CIP-3: Finalize PoW

### PR DESCRIPTION
**Pull request by fanlong**
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/cips/30)

### CODE REVIEW

**Review by Regulusyg — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/conflux-chain/cips/30)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @ChenxingLi, @dongz9, @fanlong, and @Regulusyg)

___
*[CIPs/cip-3.md, line 47 at r1](https://reviewable.io/reviews/conflux-chain/cips/30#-MIglZux-1G0Cs5TSM9a:-MIglZux-1G0Cs5TSM9b:b-m4mlt8) ([raw file](https://github.com/conflux-chain/cips/blob/bea33095290784da35f9649843c180fbb1cdb2c8/CIPs/cip-3.md#L47)):*
> ```Markdown
> 3. We should strike a balance between having enough computation power to secure
> the network after its launch and allowing mining farms (mostly right now for 
> Ethereum mining) to gain too much advantages over community miners. 
> ```

seems "allowing" should be "preventing", otherwise both goals can be achieved simultaneously
___
*[CIPs/cip-3.md, line 53 at r1](https://reviewable.io/reviews/conflux-chain/cips/30#-MIgl-FxD_0bUtTACS3f:-MIgl-FxD_0bUtTACS3g:ba3hox6) ([raw file](https://github.com/conflux-chain/cips/blob/bea33095290784da35f9649843c180fbb1cdb2c8/CIPs/cip-3.md#L53)):*
> ```Markdown
> update, we will change the PoW once to raise the memory requirement bar to exclude 
> 6G mining GPUs like P106.
> ```

6G->6GB
___
*[CIPs/cip-3.md, line 62 at r1](https://reviewable.io/reviews/conflux-chain/cips/30#-MIglqxH--1tUC-sEtX0:-MIglqxH--1tUC-sEtX1:bueagy7) ([raw file](https://github.com/conflux-chain/cips/blob/bea33095290784da35f9649843c180fbb1cdb2c8/CIPs/cip-3.md#L62)):*
> ```Markdown
> The rationales of the proposed Octopus algorihtm change are:
> 
> 1. Introduce an appropriate ammount multiplication and reminder computations that give new GPUs advantages over old GPUs.
> ```

ammount -> amount
___
*[CIPs/cip-3.md, line 66 at r1](https://reviewable.io/reviews/conflux-chain/cips/30#-MIglD4zC_0nXT5cK9wk:-MIglD4zC_0nXT5cK9wl:b-88q76a) ([raw file](https://github.com/conflux-chain/cips/blob/bea33095290784da35f9649843c180fbb1cdb2c8/CIPs/cip-3.md#L66)):*
> ```Markdown
> 2. The verification of the PoW is still as lightweight as verifying ETHASH.
> 
> 3. Set the memory requirement for mining PoW at roughtly 4.5G-5G. 
> ```

G->GB

**Review by dongz9 — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/conflux-chain/cips/30)* status: all files reviewed, 6 unresolved discussions (waiting on @ChenxingLi and @fanlong)

___
*[CIPs/cip-3.md, line 62 at r1](https://reviewable.io/reviews/conflux-chain/cips/30#-MIgqx6N7uu1rEW-F2MV:-MIgqx6N7uu1rEW-F2MW:b-vgvipt) ([raw file](https://github.com/conflux-chain/cips/blob/bea33095290784da35f9649843c180fbb1cdb2c8/CIPs/cip-3.md#L62)):*
> ```Markdown
> The rationales of the proposed Octopus algorihtm change are:
> 
> 1. Introduce an appropriate ammount multiplication and reminder computations that give new GPUs advantages over old GPUs.
> ```

reminder -> remainder
___
*[CIPs/cip-3.md, line 80 at r1](https://reviewable.io/reviews/conflux-chain/cips/30#-MIgr8pj4z6H5uNLFhcy:-MIgr8pj4z6H5uNLFhcz:b-dct8ch) ([raw file](https://github.com/conflux-chain/cips/blob/bea33095290784da35f9649843c180fbb1cdb2c8/CIPs/cip-3.md#L80)):*
> ```Markdown
> - P106: ~0.9M/s
> 
> Note that driven by profit, we expect that proprietary mining software developers will work on Conflux to optimize the miner implementation further. The mining speed therefore may improve further. 
> ```

improve further -> improve in the future

File: CIPs/cip-3.md [hunk 1]
```diff
@@ -42,40 +42,61 @@ memory.
...
 older generations, ideally proportional to its cost difference.
 
-3. Popular mining farm GPU cards, including AMD 570 4G/8G, 580 4G/8G, NVIDIA
-P106, and NVIDIA P104 should have a disadvantage comparing to newer models.
+3. We should strike a balance between having enough computation power to secure
+the network after its launch and allowing mining farms (mostly right now for 
+Ethereum mining) to gain too much advantages over community miners. 
+
+4. To address the above trade-off, we will set the minimum memory requirement of 
+participating mining to 6G at the begining of Conflux Tethys. In the future hard-fork
```
Comment (Regulusyg) [inline CIPs/cip-3.md:50]: Will Tethys start mining from 6GB or 4GB dataset?

**Review by Regulusyg — APPROVED** [review]

**Review by ChenxingLi — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/conflux-chain/cips/30)* status: all files reviewed, 7 unresolved discussions (waiting on @fanlong)

**Review by ChenxingLi — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/conflux-chain/cips/30)* status: all files reviewed, 7 unresolved discussions (waiting on @fanlong)

**Pull request merged**